File: python3/11868_v1.py
from sys import stdin
from functools import reduce

read = lambda: stdin.readline().rstrip()

N = int(read())
num_stones: list[int] = list(map(int, stdin.readline().split()))
ans = reduce(lambda acc, curr: acc ^ curr, num_stones)
if ans == 0:
    print("cubelover")
else:
    print("koosaga")

File: rust/11868_v1.rs
use std::error::Error;
use std::fmt::Write;
use std::io::{stdin, Read};

fn main() -> Result<(), Box<dyn Error>> {
    let mut output = String::new();
    let mut input = String::new();
    stdin().read_to_string(&mut input).unwrap();
    let mut input = input
        .split_ascii_whitespace()
        .map(str::parse::<i32>)
        .flatten();
    let n = input.next().unwrap();
    let num_stones: Vec<i32> = input.by_ref().take(n as usize).collect();
    let ans = num_stones
        .into_iter()
        .reduce(|acc, curr| curr ^ acc)
        .unwrap();
    if ans == 0 {
        writeln!(output, "cubelover").unwrap();
    } else {
        writeln!(output, "koosaga").unwrap();
    }
    print!("{}", output);
    Ok(())
}
